From 241d6743dc9414c2c508e710a84411ccd7015d84 Mon Sep 17 00:00:00 2001
From: Matteo Lisi <matteo.lisi@engicam.com>
Date: Wed, 22 Sep 2021 11:51:56 +0200
Subject: [PATCH] fix dtb name

---
 iMX8M/soc.mak | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 30ecefd..4e68ca0 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -137,9 +137,8 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin tee.bin
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
 
-dtbs = evk.dtb
 $(dtbs):
-	./$(DTB_PREPROC) $(PLAT)-evk.dtb $(dtbs)
+	./$(DTB_PREPROC) $(PLAT)-icore.dtb 
 
 u-boot.itb: $(dtbs)
 	./$(PAD_IMAGE) tee.bin
