From 74dbbd20a65a717e7eeae68e365cb000b380a23b Mon Sep 17 00:00:00 2001
From: Domenico Acri <domenico.acri@engicam.com>
Date: Thu, 19 Nov 2020 09:58:00 +0000
Subject: [PATCH] fixed dtb for imx8mm icore

---
 iMX8M/soc.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index b7b3986..b847921 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -133,9 +133,9 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin tee.bin
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
 
-dtbs = evk.dtb
+dtbs = icore-$(PLAT).dtb
 $(dtbs):
-	./$(DTB_PREPROC) $(PLAT)-evk.dtb $(dtbs)
+	./$(DTB_PREPROC) $(dtbs)
 
 u-boot.itb: $(dtbs)
 	./$(PAD_IMAGE) tee.bin
-- 
2.17.1

