From c0df24e3b7108577eda9fd1b514dd0658d4eb317 Mon Sep 17 00:00:00 2001
From: Domenico Acri <domenico.acri@engicam.com>
Date: Fri, 27 Mar 2020 16:13:10 +0000
Subject: [PATCH] fix dtb for icoremx8mm

---
 iMX8M/soc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index d714259..22670da 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -111,7 +111,7 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin tee.bin
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
 
-dtbs = fsl-$(PLAT)-evk.dtb
+dtbs = icore-imx8mm.dtb
 u-boot.itb: $(dtbs)
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs)
-- 
2.17.1

